src/parquet/column/reader.h

```diff
@@ -166,7 +171,7 @@ inline int64_t TypedColumnReader<DType>::ReadBatch(int batch_size, int16_t* def_
 
   // TODO(wesm): keep reading data pages until batch_size is reached, or the
   // row group is finished
-  batch_size = std::min(batch_size, num_buffered_values_);
+  batch_size = std::min(batch_size, num_buffered_values_ - num_decoded_values_);
```

Comment:
This is currently not a bug since the underlying layers  `ReadDef/RepLevels` and `ReadValues` handle cases where the requesting `batch_size` > remaining values in the current page. However, this fix corrects the logic here.